src/amo/components/FeaturedAddons/index.js

```diff
@@ -38,19 +59,39 @@ export class FeaturedAddonsBase extends React.Component {
     }
   }
 
+  loadDataIfNeeded(nextProps = {}) {
+    const { addonType: lastAddonType } = this.props;
+    const {
+      errorHandler, dispatch, loading, results,
+    } = { ...this.props, ...nextProps };
+
+    const nextParams = nextProps && nextProps.params ?
```

Comment:
I think this method could use with a few comments explaining, in plain English, what's happening. I can read through it but it took me awhile and a few re-reads to truly get what was happening. I think a comment above each section stating what it does and what it's checking for would help a lot.